core/corehttp/gateway_handler.go

```diff
@@ -372,6 +372,41 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 	}
 }
 
+type sizeReadSeeker interface {
+	Size() uint64
+
+	io.ReadSeeker
+}
+
+type sizeSeeker struct {
+	sizeReadSeeker
+}
+
+func (s *sizeSeeker) Seek(offset int64, whence int) (int64, error) {
+	if whence == io.SeekEnd && offset == 0 {
+		return int64(s.Size()), nil
+	}
+
+	return s.sizeReadSeeker.Seek(offset, whence)
+}
+
+func (i *gatewayHandler) serverFile(w http.ResponseWriter, req *http.Request, name string, modtime time.Time, content io.ReadSeeker) {
+	if sp, ok := content.(sizeReadSeeker); ok {
+		content = &sizeSeeker{
+			sizeReadSeeker: sp,
+		}
+	}
+
+	http.ServeContent(w, req, name, modtime, content)
+	//TODO: check for errors in ServeContent.. somehow
+
+	// If http.ServeContent can't figure out content size it won't write it to the
+	// responseWriter, Content-Length not being set is a good indicator of this
+	if req.Method != "HEAD" && w.Header().Get("Content-Length") == "" {
```

Comment:
This feels hacky. I think in this case, ServeContent will have already written an error status and message to the responsewriter.

Reply:
This was here before sizeReadSeeker, isn't useful now (maybe as a fallback.. removing for now)